src/Symfony/Component/Notifier/Bridge/Smsapi/SmsapiTransport.php

```diff
@@ -64,14 +64,28 @@ public function setTest(bool $test): static
 
     public function __toString(): string
     {
-        $dsn = sprintf('smsapi://%s?from=%s', $this->getEndpoint(), $this->from);
+        $dsn = sprintf('smsapi://%s', $this->getEndpoint());
+
+        if( '' !== $this->from ) {
+            $dsn .= sprintf('?from=%s', (string) $this->from);
+        }
 
         if ($this->fast) {
-            $dsn .= sprintf('&fast=%d', (int) $this->fast);
+            if( strpos($dsn, '?') === false ) 
+                $dsn .= '?';
```

Comment:
A better solution might be to build an associative array of query string attributes and then use `http_build_query` at the end instead of adding those checks for each parameter being added.